src/Components/Patient/PatientConsentRecordBlock.tsx

```diff
@@ -3,185 +3,117 @@ import {
+        >
+          <div className="flex items-center gap-4 ">
+            <div>
+              <CareIcon icon="l-file" className="text-5xl text-gray-600" />
+            </div>
+            <div className="min-w-[40%] break-all">
+              <div className="">
+                {file.name}
+                {file.extension} {file.is_archived && "(Archived)"}
               </div>
-              <div className="min-w-[40%] break-all">
-                <div className="">
-                  {file.name}
-                  {file.extension} {file.is_archived && "(Archived)"}
-                </div>
-                <div className="text-xs text-gray-700">
-                  {dayjs(file.created_date).format("DD MMM YYYY, hh:mm A")}
-                </div>
+              <div className="text-xs text-gray-700">
+                {dayjs(
+                  file.is_archived ? file.archived_datetime : file.created_date,
+                ).format("DD MMM YYYY, hh:mm A")}{" "}
+                by{" "}
+                {file.is_archived
+                  ? file.archived_by?.username
+                  : file.uploaded_by?.username}
               </div>
             </div>
-            <div className="flex shrink-0 justify-end gap-2">
-              {!file.is_archived && (
-                <ButtonV2
-                  onClick={() => previewFile(file, consentRecord.id)}
-                  className=""
-                >
-                  <CareIcon icon="l-eye" />
-                  View
-                </ButtonV2>
-              )}
-              {(file.is_archived ||
-                file?.uploaded_by?.username === authUser.username ||
+          </div>
+          <div className="flex shrink-0 flex-wrap justify-end gap-2">
+            {!file.is_archived && (
+              <ButtonV2
+                onClick={() => previewFile(file, consentRecord.id)}
+                className=""
+              >
+                <CareIcon icon="l-eye" />
+                View
+              </ButtonV2>
+            )}
+            {!file.is_archived &&
+              (file?.uploaded_by?.username === authUser.username ||
                 authUser.user_type === "DistrictAdmin" ||
                 authUser.user_type === "StateAdmin") && (
                 <ButtonV2
-                  variant={file.is_archived ? "primary" : "secondary"}
-                  onClick={() => archiveFile(file, consentRecord.id)}
+                  variant={"secondary"}
+                  onClick={() => editFile(file)}
                   className=""
                 >
-                  <CareIcon
-                    icon={file.is_archived ? "l-info-circle" : "l-archive"}
-                  />
-                  {file.is_archived ? "More Info" : "Archive"}
+                  <CareIcon icon={"l-pen"} />
+                  Rename
                 </ButtonV2>
               )}
-            </div>
+            {(file.is_archived ||
+              file?.uploaded_by?.username === authUser.username ||
```

Comment:
Move out computations as a Permission check, explaining what you're checking for in a comment